src/watcher.lisp

```diff
@@ -48,10 +48,11 @@
 (defmacro with-logged-problems (context &body body)
   `(call-with-logged-problems ',context (lambda () ,@body)))
 
-(defvar *stop-semaphore* (bt-sem:make-semaphore :name "stop-log4cl"))
+(defvar *stop-semaphore* (bt:make-lock "Stop Log4CL semaphore"))
```

Comment:
A condition variable seems better suited than just a lock. Also, if the variable holds a lock it should be called ``*stop-lock*``.